[DataGrid] Remove `dataRowIdToIdLookup` selector (#15698)
mui/mui-x: docs/data/migration/migration-data-grid-v7/migration-data-grid-v7.md

- The `apiRef.current.resize()` method was removed.
- The `<GridOverlays />` component is not exported anymore.
- `gridRowsDataRowIdToIdLookupSelector` was removed. Use `gridRowsLookupSelector` in combination with `getRowId()` API method instead.

>   ```tsx
>   // Before
>   const idToIdLookup = gridRowsDataRowIdToIdLookupSelector(apiRef);
>   ...
>   const rowId = idToIdLookup[id]
> 
>   // After
>   const rowsLookup = gridRowsLookupSelector(apiRef);
>   ...
>   const rowId = apiRef.current.getRowId(rowsLookup[id])
>   ```

Reviewer: A diff might be more readable?
```suggestion
  ```diff
  - const idToIdLookup = gridRowsDataRowIdToIdLookupSelector(apiRef);
  - const rowId = idToIdLookup[id]
  + const rowsLookup = gridRowsLookupSelector(apiRef);
  + const rowId = apiRef.current.getRowId(rowsLookup[id])
  ```
```

Author: Already aligned, based on the PR description edit you have made :)

- The `apiRef.current.resize()` method was removed.
- The `<GridOverlays />` component is not exported anymore.

> - `dataRowIdToIdLookup` selector is removed. Use `dataRowIdToModelLookup` selector in combination with `getRowId()` API method instead.

Reviewer 2: ```suggestion
- `gridRowsDataRowIdToIdLookupSelector` was removed. Use `gridRowsLookupSelector` in combination with `getRowId()` API method instead.
```
